File: Codes/0624-maximum-distance-in-arrays.cpp
/*
 * @Author: LetMeFly
 * @Date: 2025-02-19 17:37:04
 * @LastEditors: LetMeFly.xyz
 * @LastEditTime: 2025-02-19 17:46:16
 */
#ifdef _WIN32
#include "_[1,2]toVector.h"
#endif

/*
---- 
       -----


-------
    -------


--------
 ----
*/
class Solution {
public:
    int maxDistance(vector<vector<int>>& arrays) {
        int ans = 0;
        int mLeft = 100000, MRight = -100000;
        for (vector<int>& a : arrays) {
            ans = max(ans, max(a.back() - mLeft, MRight - a[0]));
            mLeft = min(mLeft, a[0]), MRight = max(MRight, a.back());
        }
        return ans;
    }
};

File: Codes/0624-maximum-distance-in-arrays.py
'''
Author: LetMeFly
Date: 2025-02-19 17:47:16
LastEditors: LetMeFly.xyz
LastEditTime: 2025-02-19 17:49:04
'''
from typing import List

class Solution:
    def maxDistance(self, arrays: List[List[int]]) -> int:
        ans = 0
        m, M = 100000, -100000
        for a in arrays:
            ans = max(ans, M - a[0], a[-1] - m)
            m, M = min(m, a[0]), max(M, a[-1])
        return ans